### PR TITLE
Add authentication guide documentation

### PR DESCRIPTION
**Pull request by stewart-gateley-boku**

### CODE REVIEW

File: AUTHENTICATION.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+Note that in addition to using this HTTP message signing scheme, it's also strongly recommended (and usually required) that the HTTP connection itself is over a TLS secured channel.
+
+# Auth signing examples / tools
+To simplify your integration, we have made several examples available, as well as tools to test your authentication implementation. These examples and tools may be downloaded from the [Boku GitHub account](https://github.com/boku-inc/boku-http-auth).
+
+# Protocol Exchange / Format
+
+## Request / Response Flow
+* With every HTTP request, the client must submit an **Authorization** header advertising their identity, and authenticating the request as from them.
+* If for any reason the submitted Authorization is deemed invalid by the sever, a HTTP **401 Unauthorized** response will be returned.
```
Comment (gmillaire) [inline AUTHENTICATION.md:16]: Typo 'sever'->'server'

File: AUTHENTICATION.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+| entity_digest | The **lower-case** hex encoded **digest** (by digest, this refers to the hash, not HMAC) of the message entity (body), using the algorithm as appropriate for this scheme. (In this version, SHA-256 is the digest used.)<br/><br/>The **raw** encoded version entity data as sent across the wire should be passed to the digest function, e.g. in the case of text a UTF-8 encoded version of an entity would have different signature to a UTF-16 encoded version of the exact same entity.<br/><br/>(**Note:** as shown above, if no entity is present in the message the trailing delimiter is still present) | Both | |
+| timestamp | The timestamp string, exactly as given in Authorization[_timestamp_] | Both | 1401143938 |
+
+## Message To Sign example - HTTP Request
+Given the following HTTP request:
+
+![Example](https://files.readme.io/18ae810-confluence-formatting-sucks-1.png)
```
Comment (gmillaire) [inline AUTHENTICATION.md:93]: Might be worth renaming the image filename to be more public-appropriate

File: AUTHENTICATION.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+For more examples of signing headers, see the [Test Vectors](#test-vectors) appendix.
+
+## Calculating the signature
+Once you have the Message to Sign and the rest of the Authorization header parameters filled out, calculating the signature is a simple matter of:
+```
+secretKey = lookupKey(Authorization[key-id])
+Authorization[signature] = hex(hmac_sha256(secretKey, messageToSign))
+```
+
+This signature should be places in the _signature_ parameter of the Authorization header to allow verification by the remote peer.
```
Comment (gmillaire) [inline AUTHENTICATION.md:170]: Typo: 'places'->'placed'

File: AUTHENTICATION.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+* Calculate the signature of the received message using the same steps as detailed in [Signature Generation Process](#signature-generation-process).
+    - Reject if the calculated signature does not equal the signature supplied in the authorization header.
+
+If the signature matches, the request or response may be processed. In the case of a request, care should be taken to check that the partner-id referenced in the header really is authorized to perform the requested operation.
+
+# Test Vectors
+
+This section contains various examples of correctly signed HTTP requests and responses, designed to allow verification of an implementation of this specification. The secret key value used for these examples is the ASCII value "secret_key_change_me".
+
+Only those types of requests which are used by the higher level application level API are required to be implemented for things to work, but to be fully compliant with this specification all of the below must produce matching signatures.
```
Comment (gmillaire) [inline AUTHENTICATION.md:197]: Not sure if this is a typo, is 'higher level application level API' supposed to be correct, or should it be 'application level API' only?

File: AUTHENTICATION.md [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+If an API makes use of HTTP headers to transmit certain important metadata about the request or response, those headers should be signed so that they cannot be altered in transit by a potential attacker.
+
+Here are some additional rules and clarifications regarding including signed-headers into the Message to Sign:
+* If a header is specified in the signed-headers field of the authorization header, it **must** appear in the request or response, otherwise an error will be raised.
+* Each unique header name must appear only once in the signed-headers list.
+* Headers are included in the Message to Sign in the order specified in the signed-headers list, **not** the order they appeared in the HTTP message.
+* Header keys are included into the Message to Sign using the **exact same case** as specified in the signed-headers field, **not** the case of the field in the HTTP request itself.
+    - Example:
+    - Authorization header contains: _signed-headers=x-foo_
+    - Authorization header contains: _signed-headers=x-foo_
```
Comment (hokubokupoku) [inline AUTHENTICATION.md:147]: Incorrect copy/paste from original article.

**Comment by stewart-gateley-boku** [comment]
All of the comments have been addressed at the new home for the documentation: https://github.com/boku-inc/boku-api-docs/blob/main/boku-api-auth/README.md

**Pull request closed without merging**